React-Projects/job-portal/src/Jobs.jsx

```diff
@@ -1,215 +1,188 @@
             <input type="checkbox" id="ruby" value="ruby" name="technology" />
-            <label for="ruby">Ruby</label>
+            <label htmlFor="ruby">Ruby</label>
           </div>
         </div>
         <h4>Location</h4>
-        <div class="filter-container">
-          <div class="filter-checkbox">
+        <div className="filter-container">
+          <div className="filter-checkbox">
             <input
               type="checkbox"
               id="ahmedabad"
               value="ahmedabad"
               name="location"
             />
-            <label for="ahmedabad">Ahmedabad</label>
+            <label htmlFor="ahmedabad">Ahmedabad</label>
           </div>
 
-          <div class="filter-checkbox">
+          <div className="filter-checkbox">
             <input type="checkbox" id="surat" value="surat" name="location" />
-            <label for="surat">Surat</label>
+            <label htmlFor="surat">Surat</label>
           </div>
 
-          <div class="filter-checkbox">
+          <div className="filter-checkbox">
             <input type="checkbox" id="goa" value="goa" name="location" />
-            <label for="goa">Goa</label>
+            <label htmlFor="goa">Goa</label>
           </div>
 
-          <div class="filter-checkbox">
+          <div className="filter-checkbox">
             <input
               type="checkbox"
               id="coimbatore"
               value="coimbatore"
               name="location"
             />
-            <label for="coimbatore">Coimbatore</label>
-          </div>
-
-          <div class="filter-checkbox">
-            <input
-              type="checkbox"
-              id="visakhapatnam"
-              value="visakhapatnam"
-              name="location"
-            />
-            <label for="visakhapatnam">Visakhapatnam</label>
+            <label htmlFor="coimbatore">Coimbatore</label>
           </div>
 
-          <div class="filter-checkbox">
+          <div className="filter-checkbox">
             <input type="checkbox" id="jaipur" value="jaipur" name="location" />
-            <label for="jaipur">Jaipur</label>
+            <label htmlFor="jaipur">Jaipur</label>
           </div>
         </div>
       </section>
-      <section class="job-list-section">
-        <div class="job-search-container">
+      <section className="job-list-section">
+        <div className="job-search-container">
           <input
-            class="job-search-input"
+            className="job-search-input"
             type="text"
             placeholder="Search for your jobs"
           />
           <h1>Jobs</h1>
         </div>
-        <div class="job-list-section-grid"></div>
+        <div className="job-list-section-grid">
+          {/* {jobData.map((job, index) => (
+            <JobCard key={index} job={job} />
+          ))} */}
+          {jobData.map((job, index) => (
+            <JobCardPython key={index} job={job} />
```

Comment:
Is key necessary? What if you don't provide it?

Reply:
key is used when creating lists of elements in React and to identify which item in array are changed. It also acts as a way identity elements in the lists. We can omit the key, but React also issue a warning if we wont provide it.